### PR TITLE
MM-17550 Added support for search term in get marketplace plugins

### PR DESCRIPTION
**Pull request by marianunez**
#### Summary
- Added search param for `getMarketplacePlugins` api endpoint. 
- Added missing test for failure case of `getMarketplacePlugins`

#### Ticket Link
[MM-17550](https://mattermost.atlassian.net/browse/MM-17550)

#### Related PR
- [Webapp PR](https://github.com/mattermost/mattermost-webapp/pull/3635)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Ran `make flow` to ensure type checking passed
- [x] Added or updated unit tests (required for all new features)
- [x] All new/modified APIs include changes to the [JavaScript driver](https://github.com/mattermost/mattermost-redux/blob/master/src/client/client4.js)

### CODE REVIEW

File: src/actions/plugins.js [hunk 1]
```diff
@@ -6,11 +6,14 @@ import {bindClientFunc} from './helpers';
 
 import {PluginTypes} from 'action_types';
 
-export function getMarketplacePlugins() {
+export function getMarketplacePlugins(searchTerm) {
```
Comment (lieut-data) [inline src/actions/plugins.js]: Not sure what pattern we've converged on for actions, but would it make sense to destructure with a default here in anticipation of future parameters? 
```suggestion
export function getMarketplacePlugins({searchTerm: ''}) {
```

File: src/client/client4.js [hunk 2]
```diff
@@ -2641,9 +2641,9 @@ export default class Client4 {
         );
     };
 
-    getMarketplacePlugins = async () => {
+    getMarketplacePlugins = async (searchTerm) => {
```
Comment (lieut-data) [inline src/client/client4.js]: Should we use the term `filter` vs. `searchTerm` for consistency with the API?

Comment (marianunez) [reply src/client/client4.js]: Done

File: src/actions/plugins.js (hunk 1 above)
Comment (marianunez) [reply src/actions/plugins.js]: Most actions in this repo pass the parameters separately so I kept that convention but I did add the default there to be cleaner.

File: src/actions/plugins.js [hunk 3]
```diff
@@ -6,11 +6,14 @@ import {bindClientFunc} from './helpers';
 
 import {PluginTypes} from 'action_types';
 
-export function getMarketplacePlugins() {
+export function getMarketplacePlugins(filter = '') {
```
Comment (ali-farooq0) [inline src/actions/plugins.js]: better to leave this as is and fix the `null` issue where we build the query params

Comment (marianunez) [reply src/actions/plugins.js]: Done

**Comment by hanzei** [comment]
@prapti Do you want to do QA for this PR pre or post merge?

**Comment by marianunez** [comment]
Merging - this would be tested in the feature branch as discussed with @prapti

**Pull request merged**